File: apps/api/src/functions/history_formatter.py
def format_history(history: dict) -> str:
    if history is None:
        return ""
    formatted_history: str = "Here's a conversation history: \n"
    for role, message in history:
        formatted_history += f"{role}: {message}\n"
    formatted_history += "Based on the above conversation, answer the following question: \n"

File: apps/worker/src/lib/functions/historyFormatter.ts
import { conversationHistoryType } from "@echo/zod/type";

export function formatHistory(
    conversations: conversationHistoryType[]
): string {
    if (!history) return "";
    let formattedHistory: string = "Here's a conversation history: \n";
    for (const conversation of conversations) {
        formattedHistory += `${conversation.role}: ${conversation.message}`;
    }
    formattedHistory += "Based on the above conversation, answer the following question: \n";
    return formattedHistory;
}